src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -295,20 +155,37 @@ case class GenericRecordReader(callFields: Set[String]) extends HtsjdkRecordRead
                   case e: Exception =>
                     fatal(
                       s"""variant $v: Genotype field ${ f.name }:
-                 |  unable to convert $x (of class ${ x.getClass.getCanonicalName }) to ${ f.typ }:
-                 |  caught $e""".stripMargin)
+                         |  unable to convert $x (of class ${ x.getClass.getCanonicalName }) to ${ f.typ }:
+                         |  caught $e""".stripMargin)
                 }
               }
             }
 
           try {
-            HtsjdkRecordReader.cast(a, f.typ)
+            var r = HtsjdkRecordReader.cast(a, f.typ)
+
+            // handle diploid
```

Comment:
Also, if this is new functionality, add a test -- or is that what testHaploid was supposed to do?